src/main/java/com/google/starfish/services/NoteService.java

```diff
@@ -11,73 +11,63 @@
- * notes table in sql
- *
+ * Service class for Notes Table
  */
 public class NoteService extends TableService {
 
+  private FavoriteNoteService favoriteNoteService = new FavoriteNoteService();
+  private MiscNoteLabelService miscNoteLabelService  = new MiscNoteLabelService();
+  private String NOTES = Table.NOTES.getSqlTable();
+
   public NoteService() {
     super(Table.NOTES);
   }
 
-  /** Gets the number of times a note has been favorited by note id */
-  public long getNumFavoritesById(DataSource pool, long noteId) throws SQLException {
-    try (Connection conn = pool.getConnection()) {
-      String stmt = 
-          "SELECT COUNT(*) AS num_favorites "
-        + "FROM starfish." + Table.FAVORITE_NOTES + " "
-        + "WHERE `note_id`=?;";
-      try (PreparedStatement selectStmt = conn.prepareStatement(stmt)) {
-        selectStmt.setLong(1, noteId);
-        ResultSet rs = selectStmt.executeQuery();
-        rs.next();
-        long numFavorites = rs.getLong("num_favorites");
-        rs.close();
-        return numFavorites;
-      }
-    }
-  }
-
   public Note[] getUploadedNotesByUserId(DataSource pool, String userId) throws SQLException {
     List<Note> notes = new ArrayList<>();
     try (Connection conn = pool.getConnection()) {
       String stmt = 
           "SELECT * "
-        + "FROM " + Table.NOTES + " "
+        + "FROM " + NOTES + " "
         + "WHERE author_id=?;";
       try (PreparedStatement upldNotesStmt = conn.prepareStatement(stmt)) {
         upldNotesStmt.setString(1, userId);
         ResultSet rs = upldNotesStmt.executeQuery();
         while (rs.next()) {
-          long noteId = rs.getLong("id");
-          String authorId = rs.getString("author_id");
-          String school = rs.getString("school");
-          String course = rs.getString("course");
-          String title = rs.getString("title");
-          String sourceUrl = rs.getString("title");
-          String pdfSource = rs.getString("pdf_source");
-          Date dateCreated = rs.getDate("date_created");
-          long numDownloads = rs.getLong("num_downloads");
-          long numFavorites = getNumFavoritesById(pool, noteId);
-
-          Note thisNote = new Note.Builder()
-                              .setId(noteId)
-                              .setAuthorId(authorId)
-                              .setRequiredLabels(school, course)
-                              .setNoteTitle(title)
-                              .setSourceUrl(sourceUrl)
-                              .setOptionalPdfSource(pdfSource)
-                              .setDateCreated(dateCreated)
-                              .setNumDownloads(numDownloads)
-                              .setNumFavorites(numFavorites)
-                              .build();
+          Note thisNote = constructNoteFromSqlResult(pool, rs);
           notes.add(thisNote);
         }
         rs.close();
         return notes.toArray(new Note[0]);
```

Comment:
Out of curiosity, why do we need to pass in a new Note object to this toArray function?

Reply:
Since this is being done inline (i.e. I'm not caching what I'm returning in a local variable first), you have to pass in the type of the array the array list is being converted into, so in this case I'm basically saying "convert this arraylist to a normal array that holds `Note` objects".
The alternative here is cache as a local variable first (which will declare the type of the array) and then return, in which case I could also do:
```
Note[] notesArray = notes.toArray();
return notesArray;
```